tests/trainer/test_trainer.py

```diff
@@ -2095,130 +2089,114 @@ def training_step(self, batch, batch_idx):
     [
         (
             dict(strategy=None, gpus=None),
-            dict(_strategy_type=None, _device_type=_AcceleratorType.CPU, num_gpus=0, num_processes=1),
+            dict(_strategy_type="single_device", _device_type=_AcceleratorType.CPU, num_gpus=0),
         ),
         (
             dict(strategy="dp", gpus=None),
-            dict(_strategy_type=None, _device_type=_AcceleratorType.CPU, num_gpus=0, num_processes=1),
+            dict(_strategy_type="ddp", _device_type=_AcceleratorType.CPU, num_gpus=0),
         ),
         (
             dict(strategy="ddp", gpus=None),
-            dict(_strategy_type=None, _device_type=_AcceleratorType.CPU, num_gpus=0, num_processes=1),
+            dict(_strategy_type="ddp", _device_type=_AcceleratorType.CPU, num_gpus=0),
         ),
         (
             dict(strategy="ddp", num_processes=2, gpus=None),
-            dict(_strategy_type=_StrategyType.DDP, _device_type=_AcceleratorType.CPU, num_gpus=0, num_processes=2),
+            dict(_strategy_type=_StrategyType.DDP, _device_type=_AcceleratorType.CPU, num_gpus=0),
         ),
         (
             dict(strategy="ddp", num_nodes=2, gpus=None),
-            dict(_strategy_type=_StrategyType.DDP, _device_type=_AcceleratorType.CPU, num_gpus=0, num_processes=1),
+            dict(_strategy_type=_StrategyType.DDP, _device_type=_AcceleratorType.CPU, num_gpus=0),
         ),
         (
             dict(strategy="ddp2", gpus=None),
-            dict(_strategy_type=None, _device_type=_AcceleratorType.CPU, num_gpus=0, num_processes=1),
+            dict(_strategy_type="ddp", _device_type=_AcceleratorType.CPU, num_gpus=0),
         ),
         (
             dict(strategy=None, gpus=1),
-            dict(_strategy_type=None, _device_type=_AcceleratorType.GPU, num_gpus=1, num_processes=1),
+            dict(_strategy_type="single_device", _device_type=_AcceleratorType.GPU, num_gpus=1),
```

Comment:
I don't like the idea of passing `"single_device"` as a training strategy. That information could be derived by the number of devices being passed to Trainer.